From ac393e10534832b3af564198992d48ab334e9ecb Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 03 Jun 2019 15:23:24 +0300
Subject: [PATCH] [SW-0] uart: introduce preboot_putc_raw() function

The preboot_putc_raw() function prints a character to the serial and the
virtual UART ports.

Change-Id: Ie8cc6ace4da89f895f96daaab91341ec565d6111
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/uart.c b/common/uart.c
index 92d729c..281f4e1 100644
--- a/common/uart.c
+++ b/common/uart.c
@@ -18,16 +18,20 @@
 	regmap_writel(val, mmPSOC_UART_0_BAUDDIV);
 }
 
+void preboot_putc_raw(const char ch)
+{
+	while (regmap_readl(mmPSOC_UART_0_STATE) & UART_SR_TXFULL)
+		;
+	regmap_writel(ch, mmPSOC_UART_0_DATA);
+	preboot_putc_virt_uart(ch);
+}
+
 void preboot_puts_raw(const char *str)
 {
 	int i;
 
-	for (i = 0; str[i] && (i < 128); i++) {
-		while (regmap_readl(mmPSOC_UART_0_STATE) & UART_SR_TXFULL)
-			;
-		regmap_writel(str[i], mmPSOC_UART_0_DATA);
-		preboot_putc_virt_uart(str[i]);
-	}
+	for (i = 0; str[i] && (i < 128); i++)
+		preboot_putc_raw(str[i]);
 }
 
 void preboot_puts(const char *str)
diff --git a/common/uart.h b/common/uart.h
index 56702b9..00712c9 100644
--- a/common/uart.h
+++ b/common/uart.h
@@ -11,6 +11,7 @@
 #define _HL_UART_H_
 
 void preboot_uart_init(unsigned int val);
+void preboot_putc_raw(const char ch);
 void preboot_puts_raw(const char *str);
 void preboot_puts(const char *str);
 int preboot_uint2hexstring(char *str, unsigned int num, bool pad);
